### ISSUE TITLE
Duplicate symbol for `flop_count` with templated code

### ISSUE THREAD

**Issue by pazner** [mfem/mfem#938]
If the header `tconfig.hpp` is included in more than one compilation unit (even with proper header guards, etc.) then the global variable `long long flop_count` results in a duplicate symbol error at link time:
```
duplicate symbol __ZN4mfem8internal10flop_countE in:
ld: 1 duplicate symbol for architecture x86_64
```

Declaring the variable as `static` will resolve this, but I'm not sure if that breaks the functionality of the flop counter. Also the definition of this variable should probably be moved within `#ifdef MFEM_COUNT_FLOPS`

**Comment by v-dobrev** [comment]
I noticed this recently as well. We should move the definition of this variable in a `.cpp` file and just have the declaration in the header. A good place for the definition is `genera/globals.cpp`, since the template code headers do not have corresponding `.cpp` files.

**Comment by pazner** [comment]
Fixed by #940
